tests/test_rpm_verifier.py

```diff
@@ -381,5 +392,32 @@ def test_main(
         )
 
         assert mock_image_processor.return_value.call_count == 1
+        generate_output_mock.assert_called_once_with(
+            [sentinel.output], fail_on_unsigned_rpms
+        )
         mock_image_processor.return_value.assert_has_calls([call("img1")])
-        mock_generate_output.assert_called_once_with([sentinel.output], True)
+
+    def test_main_fail_on_unsigned_rpm(
+        self,
+        create_generate_output_mock: MagicMock,
+        mock_image_processor: MagicMock,
+    ) -> None:
+        """Test call to rpm_verifier.py main function fails
+        when whe 'fail-unsigned' flag is used and there are unsigned RPMs
+        """
+        fail_on_unsigned_rpms: bool = True
+        create_generate_output_mock(fail=fail_on_unsigned_rpms)
+        with pytest.raises(SystemExit):
+            rpm_verifier.main(  # pylint: disable=no-value-for-parameter
+                args=[
+                    "--input",
+                    "img1",
+                    "--fail-unsigned",
+                    fail_on_unsigned_rpms,
+                    "--workdir",
+                    "some/path",
+                ],
+                obj={},
+                standalone_mode=False,
+            )
```

Comment:
We're not actually testing the failed_unsigned here, as its value does not affect the test. We're testing that we can fail based on the output of the output generation function.
```suggestion
    def test_main_fail(
        self,
        create_generate_output_mock: MagicMock,
        mock_image_processor: MagicMock,
    ) -> None:
        """Test call to rpm_verifier.py main function fails
       based on the output of the output generator
        """
        create_generate_output_mock(fail=True)
        with pytest.raises(SystemExit):
            rpm_verifier.main(  # pylint: disable=no-value-for-parameter
                args=[
                    "--input",
                    "img1",
                    "--fail-unsigned",
                    sentinel.fail_on_unsigned_rpms,
                    "--workdir",
                    "some/path",
                ],
                obj={},
                standalone_mode=False,
            )
```

Reply:
I don't mind changing it, but I think it actually points to a flaw in `generate_output`.
it is stated that the function should also `decide whether the script should fail or not`, I think this is the responsibility of the main script, and not this specific function.

Reply:
I was thinking of having it in the main script and decided against it at the time, as the alternatives did not look better to me.

You can try doing a refactor in which we do the evaluation of fail/don't-fail in main and see if it makes better sense.

Reply:
Refactored in a way that changed the failing responsibility from which fails the process from "generate_output" to the main function. The `generate_output` test is simplified as a result, and the e2e check all usecases to ensure the it fails or pass according to our needs.